src/store/user/index.js

```diff
@@ -236,9 +248,15 @@ const actions = {
     },
 
     async getUserProfile({ commit, dispatch, state }) {
+        let userType = localStorage.getItem('UserType');
+        if (userType !== "undefined" && userType !== "null") {
+            commit('set-user-type', userType);
+            return;
+        }
         try {
             const userProfile = await mayaClient.get('/auth/user');
             commit('update-user-profile', userProfile);
+            commit('set-user-type', userProfile.Type);
```

Comment:
shouldn't this commit be done from `update-user-profile`, as this is related to userProfile.